packages/interface-adapter/lib/adapter/web3/index.ts

```diff
@@ -49,8 +49,13 @@ export class Web3InterfaceAdapter implements InterfaceAdapter {
     return this.web3.eth.getAccounts();
   }
 
-  public estimateGas(transactionConfig: Transaction) {
-    return this.web3.eth.estimateGas(transactionConfig);
+  public async estimateGas(transactionConfig: Transaction) {
+    try {
```

Comment:
`try/catch` is designed for handling exceptions really, so I'm a little wary of using it for control flow. is there a better way to do this? unfortunately because there's no error handling for this case in the gas estimation call I can't just check for the promise resolution.